src/Microsoft.Identity.Web.OWIN/AppBuilderExtension.cs

```diff
@@ -238,5 +256,24 @@ public static IAppBuilder AddMicrosoftIdentityWebApp(
 
             return app.UseOpenIdConnectAuthentication(options);
         }
+
+        /// <summary>
+        /// The SDK injects 2 claims named uuid and utid into the ClaimsPrincipal, from ESTS's client_info. These represent 
+        /// the home oid and home tid and together they form the AccountId, which MSAL uses as cache key for web site 
+        /// scenarios. In case the app owner defines claims with the same name to be added to the ID Token, this creates 
+        /// a conflict with the reserved claims Id.Web injects, and it is better to throw a meaningful error. See issue 2968 for details.
+        /// </summary>
+        /// <param name="identity">The <see cref="ClaimsIdentity"/> associated to the logged-in user</param>
+        /// <param name="claimType">The claim type</param>
+        /// <param name="claimValue">The claim value</param>
+        /// <exception cref="AuthenticationException">The <see cref="ClaimsIdentity"/> contains internal claims that are used internal use by this library</exception>
+        private static void RejectInternalClaims(ClaimsIdentity identity, string claimType, string claimValue)
```

Comment:
@DOMZE I am not clear on the intent, can you please clarify? Why are we comparing the claim value? My understanding is that we should just call FindFirst and if it returns something, then throw an exception because that claim is user-provided.

Reply:
It was suggested by @bgavrilMS that if an internal claim is found in the ID Token and the value of the claim in the ID Token matches what's in the client_info, then we shouldn't throw an exception as they match.

But happy to revisit if the team is not aligned.

Reply:
@pmaytak - thanks for reviewing! That ClaimsPrincipal lifecycle is controlled by ASP.NET Core. I was thinking to avoid a situation where Id.Web injects the claims in an already modified ClaimsPrincipal and @DOMZE 's new code throws exception.

Reply:
@pmaytak has the update from @bgavrilMS made the intent clearer? Can we get this approved?